### PR TITLE
fix Search Input - composable advanced search demo

### LINKED ISSUE #9442: Bug - Search Input - composable advanced search layout broken
Screenshot 2023-08-02 at 9 41 02 AM

### PR DESCRIPTION
**Pull request by MariaAga**
**What**: Closes #9442
1. selecting a date would crash the page because of incorrect variables passed to `onChange`
2. DatePicker should not have used `isFill` since it will always use the `--pf-v5-c-date-picker__input--c-form-control--Width` width. Previously the width was ignored because input had `flex: 2;` but that css was removed in a input-group refactor

### CODE REVIEW

File: packages/react-core/src/demos/SearchInput/SearchInput.md [hunk 1]
```diff
@@ -492,7 +492,7 @@ AdvancedComposableSearchInput = () => {
                       id="datePicker"
                       style={{ width: '100%' }}
                       value={date}
-                      onChange={setDate}
+                      onChange={(e,_value) => setDate(_value)}
```
Comment (jenny-s51) [inline packages/react-core/src/demos/SearchInput/SearchInput.md:495]: Small swap - looks like `e` is the unused var here
```suggestion
                      onChange={(_e, value) => setDate(value)}
```

**Review by jenny-s51 — COMMENTED** [review]
Functionally this looks good and the date picker now works as expected @MariaAga 

Would we expect that the date picker input should stretch to the end of the panel in this case? I think visually this would make more sense.

File: packages/react-core/src/demos/SearchInput/SearchInput.md (hunk 1 above)
Comment (MariaAga) [reply packages/react-core/src/demos/SearchInput/SearchInput.md:495]: Do we use underscore and name as unused variables? I used it for value as `value` is already defined in the scope

**Comment by MariaAga** [comment]
> Would we expect that the date picker input should stretch to the end of the panel in this case? I think visually this would make more sense.

I agree that it would make more sense, but thats not how the new css works, should I open an issue in core?

File: packages/react-core/src/demos/SearchInput/SearchInput.md (hunk 1 above)
Comment (jenny-s51) [reply packages/react-core/src/demos/SearchInput/SearchInput.md:495]: Yes the PF convention is `_varName` for unused variables

**Comment by srambach** [comment]
> > > Would we expect that the date picker input should stretch to the end of the panel in this case? I think visually this would make more sense.
> > 
> > 
> > I agree that it would make more sense, but thats not how the new css works, should I open an issue in core?
> 
> cc @srambach what are your thoughts on this?

Would you want the date picker to expand across a form as well? If you let it stretch in this case, it would stretch in every case unless limited by its surroundings or possible a modifier.

My opinion is not to stretch it, because although it would then line up with the other field, the date shown within would never take up more space and it could cause visually breaking changes elsewhere. This is really a design question though... @mmenestr @andrew-ronaldson @lboehling ?

File: packages/react-core/src/demos/SearchInput/SearchInput.md (hunk 1 above)
Comment (MariaAga) [reply packages/react-core/src/demos/SearchInput/SearchInput.md]: ok, changed the names

**Comment by mmenestr** [comment]
@srambach @jenny-s51 if you're talking about this demo (want to make sure I'm looking at the right one), then I'd say no, don't stretch, I think that would look weird. I think a dropdown should mostly only be as wide as it needs to be to fit the majority of the potential menu items. In the date picker's case, it should only be basically as wide as the date within it. So I think how it is there is fine!
image

**Comment by MariaAga** [comment]
are any other changes needed?

**Review by jenny-s51 — COMMENTED** [review]
@MariaAga this is really close now - functionally everything seems to look good! Would you mind addressing the linting warnings? Otherwise should be good to merge.

As I was testing this I also noticed that our form controls look wonky in Firefox:
Screenshot 2023-11-07 at 9 50 25 AM

Opened an [issue in core](https://github.com/patternfly/patternfly/issues/6036) as it would be out of scope of this work cc @srambach

**Comment by MariaAga** [comment]
@jenny-s51  the warnings are not related to the PR and resolving them might introduce some extra logic to the component which I would rather avoid as this PR focuses on styling and demo

**Review by jenny-s51 — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Review by srambach — APPROVED** [review]
LGTM!

**Pull request merged**